src/main/java/uk/gov/hmcts/reform/civil/validation/UnavailableDateValidator.java

```diff
@@ -50,39 +51,68 @@ public boolean isValid(UnavailableDate value, ConstraintValidatorContext context
         return errors;
     }
 
+    public List<String> validateFastClaimHearing(HearingLRspec hearingLRspec) {
+        List<String> errors = new ArrayList<>();
+        if (hearingLRspec.getUnavailableDatesRequired() == YES && isFastClaimHearingNullOrEmpty(hearingLRspec)) {
+            errors.add("Details of unavailable date required");
+        }
+        if (hearingLRspec.getUnavailableDatesRequired() == YES && !isFastClaimHearingNullOrEmpty(hearingLRspec)) {
+            List<Element<UnavailableDateLRspec>> unavailabeDate
+                = hearingLRspec.getUnavailableDatesLRspec();
+            errors = dateValidationErrorCheck(unavailabeDate);
+        }
+        return errors;
+    }
+
+    public List<String> dateValidationErrorCheck(List<Element<UnavailableDateLRspec>> unavailabeDate) {
```

Comment:
It is privately used in this class if I am right.

Reply:
change incorporated